### PR TITLE
code refactorization into specific modules for each feature and expor…

### PR DESCRIPTION
**Pull request by victor-villca**
This is not a user story, but  a refactorization to our code.

### CODE REVIEW

File: src/GameOver.hs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  where
+    emptyCells = getEmptyCells board
+    noMergePossible = not (any id (map (\index -> verifyDirections index board) (indices board)))
+
+verifyDirections :: (Int, Int) -> Board -> Bool
+verifyDirections index board = 
+  verifyUp index board || verifyDown index board || verifyLeft index board || verifyRight index board
+
+verifyUp :: (Int, Int) -> Board -> Bool
+verifyUp (x, y) board = x > 0 && board ! (x, y) == board ! (x-1, y)
```
Comment (DeidamiaFuentes) [inline src/GameOver.hs:19]: No instance for ‘Eq Cell’ arising from a use of ‘==’
• In the second argument of ‘(&&)’, namely
    ‘board ! (x, y) == board ! (x + 1, y)’
  In the expression:
    x < n - 1 && board ! (x, y) == board ! (x + 1, y)
  In an equation for ‘verifyDown’:
      verifyDown (x, y) board
        = x < n - 1 && board ! (x, y) == board ! (x + 1, y)typecheck(-Wdeferred-type-errors)

File: src/GameOver.hs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+verifyDown :: (Int, Int) -> Board -> Bool
+verifyDown (x, y) board = x < n-1 && board ! (x, y) == board ! (x+1, y)
```
Comment (DeidamiaFuentes) [inline src/GameOver.hs:22]: No instance for ‘Eq Cell’ arising from a use of ‘==’
• In the second argument of ‘(&&)’, namely
    ‘board ! (x, y) == board ! (x + 1, y)’
  In the expression:
    x < n - 1 && board ! (x, y) == board ! (x + 1, y)
  In an equation for ‘verifyDown’:
      verifyDown (x, y) board
        = x < n - 1 && board ! (x, y) == board ! (x + 1, y)typecheck(-Wdeferred-type-errors)

File: src/GameOver.hs [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+verifyLeft :: (Int, Int) -> Board -> Bool
+verifyLeft (x, y) board = y > 0 && board ! (x, y) == board ! (x, y-1)
```
Comment (DeidamiaFuentes) [inline src/GameOver.hs:25]: • No instance for ‘Eq Cell’ arising from a use of ‘==’
• In the second argument of ‘(&&)’, namely
    ‘board ! (x, y) == board ! (x, y - 1)’
  In the expression: y > 0 && board ! (x, y) == board ! (x, y - 1)
  In an equation for ‘verifyLeft’:
      verifyLeft (x, y) board
        = y > 0 && board ! (x, y) == board ! (x, y - 1)typecheck(-Wdeferred-type-errors)

File: src/GameOver.hs [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+verifyRight :: (Int, Int) -> Board -> Bool
+verifyRight (x, y) board = y < n-1 && board ! (x, y) == board ! (x, y+1)
```
Comment (DeidamiaFuentes) [inline src/GameOver.hs:28]: • No instance for ‘Eq Cell’ arising from a use of ‘==’
• In the second argument of ‘(&&)’, namely
    ‘board ! (x, y) == board ! (x, y + 1)’
  In the expression:
    y < n - 1 && board ! (x, y) == board ! (x, y + 1)
  In an equation for ‘verifyRight’:
      verifyRight (x, y) board
        = y < n - 1 && board ! (x, y) == board ! (x, y + 1)typecheck(-Wdeferred-type-errors)

File: src/Movement.hs [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+import Random
+import GameOver
+
+
+-- Shifts all the non empty cells in a row to the left
+-- Empty cells are moved to the right side of the row
+shiftRow :: [Cell] -> [Cell]
+shiftRow row = filtered ++ padding
+  where
+    filtered = filter (/= Empty) row
```
Comment (DeidamiaFuentes) [inline src/Movement.hs:15]: • No instance for ‘Eq Cell’ arising from a use of ‘/=’
• In the expression: (/=)
  In the first argument of ‘filter’, namely ‘(/= Empty)’
  In the expression: filter (/= Empty) rowtypecheck(-Wdeferred-type-errors)

File: src/Utils.hs [hunk 6]
```diff
@@ -15,5 +24,34 @@ boldText offset textPic = pictures
...
+
+
+    -- Function to create a new empty board
+createEmptyBoard :: Int -> Board
+createEmptyBoard a = array indexRange $ zip (range indexRange) (cycle [Empty])
+  where
+    indexRange = ((0, 0), (a - 1, a - 1))
+
+getEmptyCells :: Board -> [(Int, Int)]
+getEmptyCells board = filter (\coord -> board ! coord == Empty) (indices board)
```
Comment (DeidamiaFuentes) [inline src/Utils.hs:57]: • No instance for ‘Eq Cell’ arising from a use of ‘==’
• In the expression: board ! coord == Empty
  In the first argument of ‘filter’, namely
    ‘(\ coord -> board ! coord == Empty)’
  In the expression:
    filter (\ coord -> board ! coord == Empty) (indices board)typecheck(-Wdeferred-type-errors)

**Review by DeidamiaFuentes — APPROVED** [review]
Everything looks fine

**Review by AngheloZambrana — APPROVED** [review]
Good job

**Review by MayerliSantander — APPROVED** [review]

**Pull request merged**